DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/zero_copy_data_transfer/src/restricted_usm_kernel.hpp

```diff
@@ -49,32 +49,27 @@ class Producer;
 class Consumer;
 
 // pipes
-constexpr size_t kPipeDepth = 64;
 template <typename T>
-using ProducePipe = pipe<class ProducePipeClass, T, kPipeDepth>;
```

Comment:
Why are you removing the explicit specification of the pipe depth here?  Does this have any impact on functionality or performance?

Reply:
No impact on functionality or performance. The compiler can do a better job of sizing here rather than the me.